### PR TITLE
Add tenantDomain to StoryCreatedEventPayload

### PR DESCRIPTION
**Pull request by cristiandean**
## What does this PR do?
When you have multiple Talk instances running, it seems to make sense to specify from which tenant a given event was sent.

This PR add `tenantDomain`  to `StoryCreatedCoralEventPayload`.

## How do I test this PR?
Create a webhook, access some story and look at the request body.

### CODE REVIEW

**Comment by cristiandean** [comment]
I've been thinking that maybe the tenant domain could be placed in the header. If you agree I can change.

File: src/core/server/services/stories/index.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ export async function findOrCreate(
   scraper: ScraperQueue,
   now = new Date()
 ) {
-  let siteID = null;
+  let siteID = "";
```
Comment (wyattjoh) [inline src/core/server/services/stories/index.ts]: I would revert this change here, and instead below in the call to `StoryCreatedCoralEvent.publish` you use `siteID: story.siteID` instead.

File: src/core/server/services/stories/index.ts [hunk 2]
```diff
@@ -94,6 +94,8 @@ export async function findOrCreate(
     StoryCreatedCoralEvent.publish(broker, {
       storyID: story.id,
       storyURL: story.url,
+      tenantID: tenant.id,
```
Comment (wyattjoh) [inline src/core/server/services/stories/index.ts]: I want to move the `tenantID` out of the publish call, I'll be submitting a PR shortly that will add the tenantID to the payload instead at a higher level.

**Review by wyattjoh — CHANGES_REQUESTED** [review]

**Comment by wyattjoh** [comment]
I've gone ahead and added the `tenantID` to the payload for all webhook payloads now and added to the webhook documentation. Feel free to comment on those changes!

File: src/core/server/services/stories/index.ts (hunk 2 above)
Comment (wyattjoh) [reply src/core/server/services/stories/index.ts]: Note that this is now available on the event object itself (not the `data` member).

**Comment by cristiandean** [comment]
@wyattjoh I had made a few more changes but I hadn't submitted.

In addition to tenantID, I think it would be good if we also sent tenantDomain. For example, we have several instances of talk running often on different servers and databases. We need to know from which of these instances we received the event.

If we have the domain of the tenant within the webhook we can run queries on it on its _/api/graphql_ route.


What do you think?

**Comment by wyattjoh** [comment]
I agree that adding the tenant domain could be helpful here as well. If you can you push/merge your additional changes, I can add the `tenantDomain` using a similar method to `tenantID`!

File: src/core/server/services/stories/index.ts (hunk 2 above)
Comment (cristiandean) [reply src/core/server/services/stories/index.ts]: Great! I already removed it from there.

**Comment by cristiandean** [comment]
I ended up resolving the siteID issue and removing the tenantID from StoryCreatedCoralEvent.

I had also put tenantDomain into the data, but what you said makes perfect sense, it should be one level up and available from all hooks.

**Comment by cristiandean** [comment]
I also added tenantDomain to the webhook schema.

**Review by wyattjoh — APPROVED** [review]
Looks awesome!
